pkg/daemon/update.go

```diff
@@ -78,13 +82,36 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 			IgnoreDaemonsets:   true,
 		})
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "Draining non-daemonsets")
 		}
-		glog.V(2).Infof("Node successfully drained")
+		glog.V(2).Infof("Node successfully drained (non-daemonsets)")
 	}
 
-	// reboot. this function shouldn't actually return.
-	return dn.reboot(fmt.Sprintf("Node will reboot into config %v", newConfigName))
+	err = dn.queueReboot(fmt.Sprintf("Node will reboot into config %v", newConfigName))
+	if err != nil {
+		return err
+	}
+
+	// And now this time without IgnoreDaemonSets, which should as a side
+	// effect kill us and release the reboot lock, allowing the queued
+	// reboot to proceed.
+	if !onceFromMode {
+		err = drain.Drain(dn.kubeClient, []*corev1.Node{node}, &drain.DrainOptions{
```

Comment:
Hmm, but how will Kubernetes know that it's OK to re-schedule us on that node if it doesn't ignore DaemonSets? We'd need some other service to uncordon the node once it's back up, right?

Reply:
(To clarify, the MCD currently is the one that uncordons the node once it's back up and completes the update).

Reply:
Ah, right...yeah then to complete this story we'd need to push a systemd unit to the host that does that.  Which gets into interesting issues since we'd need to stash the kube creds.

We could move the uncordoning into the operator?  Actually, we could also move the drain there.  Have the MCD signal that it's queued a reboot successfully via node annotation.

Really it feels like there's some duplication in role here between kubelet and MCD.  If kubelet communicates boot ID to master, then we can update node annotations from that.